codegen/module_system.go

```
@@ -65,44 +63,136 @@ type StructMeta struct {
-
-	ClientStubs []ClientStub
-
-	TestServiceName string // The service module that mounts the endpoint
+// FixtureBlob is map[string]interface{} that implements default string
+// used for headers and (http | tchannel) request/response
+type FixtureBlob map[string]interface{}
+
+// String convert FixtureBlob to string
+func (fb *FixtureBlob) String() string {
+	str, err := json.Marshal(fb)
+	if err != nil {
+		panic(err)
+	}
+	return string(str)
+}
+
+// BodyType can either be `json` or `string`
+type BodyType string
+
+// ProtocalType can either be `http` or `tchannel`
+type ProtocalType string
+
+// HTTPMethodType can only be valid http method
+type HTTPMethodType string
+
+// FixtureBody is used to create http body in test fixtures
+type FixtureBody struct {
+	BodyType   BodyType     `json:"bodyType,omitempty"`
+	BodyString string       `json:"bodyString,omitempty"` // set BodyString if response body is string
+	BodyJSON   *FixtureBlob `json:"bodyJson,omitempty"`   // set BodyJSON if response body is object
+}
+
+// String convert FixtureBody to string
+func (fb *FixtureBody) String() string {
+	switch fb.BodyType {
+	case "string":
+		return fb.BodyString
+	case "json":
+		if fb.BodyJSON == nil {
+			panic(errors.New("invalid http body type"))
+		}
+		return fb.BodyJSON.String()
+	default:
+		panic(errors.New("invalid http body type"))
+	}
+}
+
+// FixtureHTTPResponse is test fixture for http response
+type FixtureHTTPResponse struct {
+	StatusCode int          `json:"statusCode"`
+	Body       *FixtureBody `json:"body,omitempty"`
+}
+
+// FixtureResponse is test fixture for client/endpoint response
+type FixtureResponse struct {
+	ResponseType     ProtocalType         `json:"responseType"`
+	HTTPResponse     *FixtureHTTPResponse `json:"httpResponse,omitempty"`
+	TChannelResponse FixtureBlob          `json:"tchannelResponse,omitempty"`
+}
+
+// Body returns the string representation of FixtureResponse
```

Let' leave a comment on why this func is panic instead of returning error.